geode-core/src/main/java/org/apache/geode/management/internal/functions/CacheRealizationFunction.java

```diff
@@ -78,24 +79,44 @@
   public void execute(FunctionContext<List> context) {
     AbstractConfiguration cacheElement = (AbstractConfiguration) context.getArguments().get(0);
     CacheElementOperation operation = (CacheElementOperation) context.getArguments().get(1);
-    RemoteInputStream jarStream = (RemoteInputStream) context.getArguments().get(2);
 
-    InternalCache cache = (InternalCache) context.getCache();
-    try {
-      if (operation == CacheElementOperation.GET) {
+    // for get operation, caller is expecting RuntimeInfo
+    if (operation == CacheElementOperation.GET) {
+      try {
+        InternalCache cache = (InternalCache) context.getCache();
         context.getResultSender().lastResult(executeGet(context, cache, cacheElement));
-      } else {
+      } catch (CacheClosedException e) {
```

Comment:
I am not sure the `CacheClosedException` in the JIRA is thrown this function. This function does not call `ResultSender.sendException()` to the caller. How can the caller `LocatorClusterManagementService` get the `CacheClosedException` and then `ClassCastException`?

Reply:
> I am not sure the `CacheClosedException` in the JIRA is thrown this function. This function does not call `ResultSender.sendException()` to the caller. How can the caller `LocatorClusterManagementService` get the `CacheClosedException` and then `ClassCastException`?

the exception is thrown by `context.getCache()` method, (https://github.com/apache/geode/blob/ba8a5528f462c86104deb531b18245db64efe324/geode-core/src/main/java/org/apache/geode/internal/cache/execute/FunctionContextImpl.java#L109). Before the fix, the exception is thrown and caught by the function executor and then it directly gave the exception back to the caller, but caller is expecting a `RuntimeInfo` object and tried to cast the exception to `RuntimeInfo`, thus the error.